packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/useCreditCard.js

```diff
@@ -392,13 +388,25 @@ export const useCreditCard = props => {
     useEffect(() => {
         try {
             if (shouldSubmit) {
-                setStepNumber(1);
-                if (isBillingAddressSame) {
-                    setShippingAddressAsBillingAddress();
+                /**
+                 * Validate billing address fields and only process with
+                 * submit if there are no errors.
```

Comment:
Update this comment explaining why we are manually invoking validation (no form submit) and also why we can't just validate on blur (a user can skip the form entirely and just click review)